### ISSUE TITLE
Particles flickering

### ISSUE THREAD

**Issue by Baransu** [Netflix/vizceral#62]
When we have connection between two nodes and one of them is another color (warrning, error) there is flickering between layers. There is probably the same thing with normal particles but because of the same neutral color it's not visible.

This may happend because of texture overlay on the same axis.

Here is an example: https://youtu.be/-AvscsVSyck

**Comment by Baransu** [comment]
Here is example data when this happen with Vizceral `4.4.0`: https://gist.github.com/Baransu/a6a77996e758c523e2af0875a9aed9c9

**Comment by jrsquared** [comment]
Yep, looks exactly like because the connections are on the same axis. We don't have any bi-directional connections that have traffic at the same time, so that's why we've never run into this.

https://github.com/Netflix/vizceral/blob/master/src/base/connectionView.js#L222 should probably have some logic to determine directionality and change depth(`z`) based on that.

One easy (hacky?) way to do it would be to use alphabetical ordering of name to determine how to set z (`if source.name > target.name: z += 1`), so that if the connection runs the other way, it will have a different depth.

Try that and see how it feels. If you think of a better way, happy for suggestions too.

**Comment by Baransu** [comment]
Changing depth based on alphabetical ordering is not that bad. I was thinking how we can use `.isEntryNode()` for that but this would only apply for connection to/from `entryNode` and we can have bi-directional connection between other nodes to. I would go with alphabetical ordering.

**Comment by Baransu** [comment]
I'm surprised but this doesn't fix flickering 🤔

**Comment by jrsquared** [comment]
Sorry I dropped this one. I honestly thought that would fix it. If you still are having this issue, I can take a look sometime this week.

**Comment by remiphilippe** [comment]
@Baransu out of curiosity, did you run into and issue where things overlap and keep on loading (only with bidir traffic, removing the second direction fixes the problem)?

![image](https://user-images.githubusercontent.com/1027614/27264899-be64fc00-543e-11e7-9b33-2052bce0dc5a.png)

**Comment by Baransu** [comment]
You have region loading because you don't have `entryNode` specified probablly.

I had flickering with bidir connection when one of them was in different color which multiplies flickering effect because you see how layers are overlaping each other.

**Comment by remiphilippe** [comment]
Good catch thanks, I typed `entryPoint` instead of `entryNode`, still shows loading but there could be another error. I see the same flickering with color though not major in my case.

![image](https://user-images.githubusercontent.com/1027614/27315595-5c485016-552f-11e7-9d0a-9e861ec37bdf.png)
